### ISSUE TITLE
[Android Studio] Crash IDE due to duplicate annotation

### ISSUE THREAD

**Issue by proninyaroslav** [Gericop/Android-Support-Preference-V7-Fix#40]
Stack trace:

```
Duplicate annotation 'android.support.annotation.IntDef'; for signature: 'com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags' in the file /home/yaroslav/AndroidStudioProjects/LibreTorrent/app/build/intermediates/exploded-aar/com.takisoft.fix/preference-v7/24.2.0.0/annotations.zip!/com/takisoft/fix/support/v7/preference/annotations.xml
java.lang.Throwable
    at com.intellij.diagnostic.LogMessageEx.error(LogMessageEx.java:136)
    at com.intellij.codeInsight.ExternalAnnotationsManagerImpl.duplicateError(ExternalAnnotationsManagerImpl.java:709)
    at com.intellij.codeInsight.BaseExternalAnnotationsManager$DataParsingSaxHandler.endElement(BaseExternalAnnotationsManager.java:468)
    at org.apache.xerces.parsers.AbstractSAXParser.endElement(Unknown Source)
    at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanEndElement(Unknown Source)
    at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
    at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
    at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
    at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
    at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
    at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
    at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
    at org.apache.xerces.jaxp.SAXParserImpl.parse(Unknown Source)
    at com.intellij.codeInsight.BaseExternalAnnotationsManager.getDataFromFile(BaseExternalAnnotationsManager.java:178)
    at com.intellij.codeInsight.BaseExternalAnnotationsManager.doCollect(BaseExternalAnnotationsManager.java:219)
    at com.intellij.codeInsight.BaseExternalAnnotationsManager.collectExternalAnnotations(BaseExternalAnnotationsManager.java:146)
    at com.intellij.codeInsight.BaseExternalAnnotationsManager.findExternalAnnotations(BaseExternalAnnotationsManager.java:123)
    at org.jetbrains.android.inspections.lint.LombokPsiParser.getAnnotations(LombokPsiParser.java:430)
    at org.jetbrains.android.inspections.lint.LombokPsiParser.access$900(LombokPsiParser.java:57)
    at org.jetbrains.android.inspections.lint.LombokPsiParser$ResolvedPsiMethod.getAnnotations(LombokPsiParser.java:559)
    at com.android.tools.lint.checks.CallSuperDetector.getRequiredSuperMethod(CallSuperDetector.java:164)
    at com.android.tools.lint.checks.CallSuperDetector.checkCallSuper(CallSuperDetector.java:117)
    at com.android.tools.lint.checks.CallSuperDetector.access$000(CallSuperDetector.java:53)
    at com.android.tools.lint.checks.CallSuperDetector$1.visitMethodDeclaration(CallSuperDetector.java:105)
    at com.android.tools.lint.client.api.JavaVisitor$DispatchVisitor.visitMethodDeclaration(JavaVisitor.java:1066)
    at lombok.ast.MethodDeclaration.accept(MethodDeclaration.java:226)
    at lombok.ast.NormalTypeBody.accept(NormalTypeBody.java:57)
    at lombok.ast.ClassDeclaration.accept(ClassDeclaration.java:239)
    at lombok.ast.CompilationUnit.accept(CompilationUnit.java:83)
    at com.android.tools.lint.client.api.JavaVisitor.visitFile(JavaVisitor.java:274)
    at com.android.tools.lint.client.api.LintDriver.checkIndividualJavaFiles(LintDriver.java:1571)
    at com.android.tools.lint.client.api.LintDriver.runFileDetectors(LintDriver.java:1049)
    at com.android.tools.lint.client.api.LintDriver.checkProject(LintDriver.java:908)
    at com.android.tools.lint.client.api.LintDriver.analyze(LintDriver.java:436)
    at com.android.tools.lint.client.api.LintDriver.analyze(LintDriver.java:377)
    at org.jetbrains.android.inspections.lint.AndroidLintExternalAnnotator.doAnnotate(AndroidLintExternalAnnotator.java:168)
    at org.jetbrains.android.inspections.lint.AndroidLintExternalAnnotator.doAnnotate(AndroidLintExternalAnnotator.java:67)
    at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:218)
    at com.intellij.codeInsight.daemon.impl.ExternalToolPass.access$500(ExternalToolPass.java:44)
    at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:160)
    at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:337)
    at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:327)
    at com.intellij.util.ui.update.MergingUpdateQueue$3.run(MergingUpdateQueue.java:271)
    at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:286)
    at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:244)
    at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:234)
    at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
    at com.intellij.util.Alarm$Request$1.run(Alarm.java:352)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:298)
    at com.intellij.util.concurrency.QueueProcessor$RunnableConsumer.consume(QueueProcessor.java:295)
    at com.intellij.util.concurrency.QueueProcessor$2$1.run(QueueProcessor.java:110)
    at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
    at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:107)
    at com.intellij.util.concurrency.QueueProcessor$2.consume(QueueProcessor.java:104)
    at com.intellij.util.concurrency.QueueProcessor$3$1.run(QueueProcessor.java:215)
    at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
    at com.intellij.util.concurrency.QueueProcessor$3.run(QueueProcessor.java:212)
    at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:366)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
    at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:55)

```

It's generated by Studio:

``` XML
<?xml version="1.0" encoding="UTF-8"?>
<root>
  <item name="com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags">
    <annotation name="android.support.annotation.IntDef">
      <val name="flag" val="true" />
      <val name="value" val="{com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_BETWEEN, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_BEFORE_FIRST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_AFTER_LAST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_BETWEEN, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_BEFORE_FIRST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_AFTER_LAST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PADDING_CHILD, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PADDING_PARENT}" />
    </annotation>
    <annotation name="android.support.annotation.IntDef">
      <val name="value" val="{com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_DEFAULT, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_NONE}" />
    </annotation>
  </item>
  <item name="com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers void applyDividerPreference(android.support.v7.widget.RecyclerView, int) 1">
    <annotation name="android.support.annotation.IntDef">
      <val name="flag" val="true" />
      <val name="value" val="{com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_BETWEEN, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_BEFORE_FIRST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_AFTER_LAST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_BETWEEN, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_BEFORE_FIRST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_AFTER_LAST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PADDING_CHILD, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PADDING_PARENT}" />
    </annotation>
    <annotation name="android.support.annotation.IntDef">
      <val name="value" val="{com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_DEFAULT, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_NONE}" />
    </annotation>
  </item>
  <item name="com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers void setDividerPreferences(int) 0">
    <annotation name="android.support.annotation.IntDef">
      <val name="flag" val="true" />
      <val name="value" val="{com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_BETWEEN, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_BEFORE_FIRST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_CATEGORY_AFTER_LAST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_BETWEEN, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_BEFORE_FIRST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PREFERENCE_AFTER_LAST, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PADDING_CHILD, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_PADDING_PARENT}" />
    </annotation>
    <annotation name="android.support.annotation.IntDef">
      <val name="value" val="{com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_DEFAULT, com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers.DIVIDER_NONE}" />
    </annotation>
  </item>
</root>
```

**Comment by Gericop** [comment]
What do your gradle files look like?

**Comment by proninyaroslav** [comment]
I don't quite understand the question. I use the last version of gradle, which download Studio. I tried a sample app, but there I encountered the same problem.

**Comment by Gericop** [comment]
I'm curious about your targetSdkVersion, the build tool version and the dependencies. This problem happens if you use different versions of dependent libraries and/or the build tool is old (and buggy, like the v23).

**Comment by proninyaroslav** [comment]
```
compileSdkVersion 24
buildToolsVersion "24.0.1"
targetSdkVersion 24
```

dependencies:
```GRADLE
compile 'com.github.aakira:expandable-layout:1.5.1@aar'
compile 'com.github.clans:fab:1.6.3'
def ANDROID_SUPPORT_VERSION='24.2.0'
compile 'com.android.support:support-v4:' + ANDROID_SUPPORT_VERSION
compile 'com.android.support:appcompat-v7:' + ANDROID_SUPPORT_VERSION
compile 'com.android.support:design:' + ANDROID_SUPPORT_VERSION
compile 'com.android.support:cardview-v7:' + ANDROID_SUPPORT_VERSION
```

**Comment by Gericop** [comment]
Even with your dependencies copied, it works perfectly for me. Try clean & rebuild.

**Comment by proninyaroslav** [comment]
I cleared project and .gradle dir, but it didn't help. Also I tried a sample app, but there I encountered the same problem.

**Comment by Gericop** [comment]
What's your Android Studio and gradle version?

**Comment by proninyaroslav** [comment]
Studio 2.1.3, gradle 3.0 (prev. 2.14.1), gradle Android plugin 2.1.3

**Comment by Gericop** [comment]
Is this a library or an application project?

**Comment by proninyaroslav** [comment]
This is an app.

**Comment by Gericop** [comment]
Well, I'm out of ideas. It seems to be working for everyone with probably the same configuration. If there are no .jar files in your lib or libs folder, nothing I can do. This seems like a local problem. Try copying only the sources (and resources of course) to a new project, then gradually upgrade the .gradle files to see, what causes the problem.

**Comment by proninyaroslav** [comment]
Ok, I'll try to run it on another computer.

**Comment by anpaza** [comment]
Same problem happens here.
Duplicate annotation 'android.support.annotation.IntDef'; for signature: 'com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags' in the file /home/zap/electronics/projects/xpd/android/app/build/intermediates/exploded-aar/com.takisoft.fix/preference-v7/25.0.1.0/annotations.zip!/com/takisoft/fix/support/v7/preference/annotations.xml

And since it's a opensource project, I can give you a direct link to the whole project:
https://sourceforge.net/p/xpd-ebike/code/HEAD/tree/trunk/android/

Just open it in Android Studio, and this message appears in the status bar.

**Comment by Gericop** [comment]
The problem is that you use different versions of support library components. I navigated into your `app/build.gradle` file and there are a couple problems. First of all, to use v25.x.y, you need buildToolsVersion and compileSdkVersion set to 25 or above. Another problem is that you should use the same version of support lib components and since the current version of the library fix is meant for 25.0.1, instead of
```
compile 'com.android.support:appcompat-v7:24.2.1'
compile 'com.android.support:design:24.2.1'
compile 'com.android.support:support-v4:24.2.1'
compile 'com.android.support:recyclerview-v7:24.2.1'
```
use
```
compile 'com.android.support:appcompat-v7:25.0.1'
compile 'com.android.support:design:25.0.1'
compile 'com.android.support:support-v4:25.0.1'
compile 'com.android.support:recyclerview-v7:25.0.1'
```

Please do keep in mind that I'll make the 25.1.0 compatible version available soon, so if you want to upgrade to that later, you'll need to change the other components' support lib version too.

**Comment by anpaza** [comment]
Thank you for help! This is my first android program ever, so no wonder I still do childish errors :)
I disabled automatic updates because it had no sources for API Level 25. Now it has.
It took a while to update & test, but everything seems fine now.

**Comment by Mygod** [comment]
Oh FYI sources for API level 25 are now available.

**Comment by anpaza** [comment]
I was wrong. Still get same error from time to time, can't understand what causes it, but happens pretty often.
Same message:

0:15:55 Duplicate annotation 'android.support.annotation.IntDef'; for signature: 'com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags' in the file /home/zap/electronics/projects/xpd/android/xpd/build/intermediates/exploded-aar/com.takisoft.fix/preference-v7/25.0.1.0/annotations.zip!/com/takisoft/fix/support/v7/preference/annotations.xml: Duplicate annotation 'android.support.annotation.IntDef'; for signature: 'com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags' in the file /home/zap/electronics/projects/xpd/android/xpd/build/intermediates/exploded-aar/com.takisoft.fix/preference-v7/25.0.1.0/annotations.zip!/com/takisoft/fix/support/v7/preference/annotations.xml

Switched all libs to 25.0.1, but doesn't help. The updated sources are still available by the link above.

**Comment by Gericop** [comment]
Did you try cleaning the project before building? Might be some residual files in cache.

**Comment by anpaza** [comment]
I tried to clean the project, then restart the Android Studio and the error appears again.
I even tried to remove 'build' subdirectories from everywhere in project tree (while Android Studio was not running). Nothing helps. I get exception in "Android Support" IDE plugin with the following content:

The Comments tab:
Duplicate annotation 'android.support.annotation.IntDef' [...] same like above

The Details tab:
Duplicate annotation 'android.support.annotation.IntDef'; for signature: 'com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags' in the file /home/zap/electronics/projects/xpd/android/xpd/build/intermediates/exploded-aar/com.takisoft.fix/preference-v7/25.0.1.0/annotations.zip!/com/takisoft/fix/support/v7/preference/annotations.xml
java.lang.Throwable
	at com.intellij.diagnostic.LogMessageEx.error(LogMessageEx.java:136)
	at com.intellij.codeInsight.ExternalAnnotationsManagerImpl.duplicateError(ExternalAnnotationsManagerImpl.java:709)
	at com.intellij.codeInsight.BaseExternalAnnotationsManager$DataParsingSaxHandler.endElement(BaseExternalAnnotationsManager.java:468)
	at org.apache.xerces.parsers.AbstractSAXParser.endElement(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanEndElement(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl.parse(Unknown Source)
...

The Attachment tab:
(the content of your annotations.xml from com.takisoft.fix/preference-v7/25.0.1.0/annotations.zip)

Unfortunately, I know little about annotations, so I can't understand what's wrong here :(
You can try to download the project from the link above (it does not need any extras), I think the problem will reproduce.

**Comment by Gericop** [comment]
I downloaded a snapshot from [your repo](https://sourceforge.net/p/xpd-ebike/code/HEAD/tree/trunk/android/), extracted it, then Android Studio -> File -> New -> Import project..., and selected the android folder. After finishing the import, I opened up an API 23 emulator and hit *run*, and it did run. So I can't reproduce the error, sorry.

What version of Java do you use?

**Comment by anpaza** [comment]
Android Studio 2.2.3
javac 1.8.0_111

**Comment by Gericop** [comment]
Is that for `javac -version` (mind the `c` at the end)?
Anyways, it works for me with your project. What if you try and import it as a new project from the version controlled source?

**Comment by anpaza** [comment]
The above is result of "javac --version" command. My OS is Fedora 25.
javac is a symlink to the javac binary from package java-1.8.0-openjdk-devel-1.8.0.111-3.b16.fc24.x86_64
---
Yes, I managed to reproduce the problem with a fresh svn checkout. But the problem appeared only after some manipulations, not instantly.
The following sequence did it:
1. Checkout a fresh copy somewhere
2. Run Android Studio, open it. It says gradle is not configure, I hit OK.
3. Open some source, I open xpd/ActivityMain.
4. Build -> Clean Project
5. Quit Android Studio
6. Launch Android Studio, wait until it finishes reloading everything
7. Build -> Rebuild Project
8. Quit Android Studio
9. Relaunch Android Studio
10. Repeat steps 6-10 a few times, and the problem appears during project load after Android Studio launch. A popup shows in top-right corner, and a red exclamation sign icon flashes to the left of "Event Log" button in lower-right corner.

After that the error happens any time you reload Android Studio or do clean, rebuild etc...

Well, it looks like this is an Android Studio and/or Gradle bug, so I think we should drop this...

**Comment by Gericop** [comment]
Interesting. I met the same problem earlier in a different project, but it was caused by the different versions of the support libraries resulting in multiple versions of the same class, so the Android Studio (or rather Gradle) couldn't merge them.

One thing you can do is to check out this project and modify the [`PreferenceFragmentCompatDividers`](https://github.com/Gericop/Android-Support-Preference-V7-Fix/blob/master/preference-v7/src/main/java/com/takisoft/fix/support/v7/preference/PreferenceFragmentCompatDividers.java) class so that you remove the `IntDef` part. It has only semantical meaning but it helps one in the IDE by allowing auto-complete on the divider settings. Then you can add the dependency to that project instead of the jcenter one. Of course, this also means that you'd have to do the update / merge process when a new version is available.

Or you can try something like [this](http://stackoverflow.com/a/32337164) with the configuration exclusions, but I'm not sure it would fix it and not just mess up the whole thing.

**Comment by anpaza** [comment]
I understand I can patch your library to remove annotations, but usually I don't like such 'fixes'.

I also tried the suggestion with Gradle, notably I've changed the line requesting compiling of com.takisoft.fix to:

compile ('com.takisoft.fix:preference-v7:25.0.1.0') {
        exclude module: 'support-annotations'
}

but unfortunately, this does not work. The above was done by intuition and some reading of Gradle docs, but Gradle is still pretty much dark magic to me :(

**Comment by Gericop** [comment]
@anpaza What happens if you try *File* -> *Invalidate Caches / Restart...* ?

**Comment by BuhTigRunner** [comment]
I have the same problem.  Android Studio 2.2.3. 
File -> Invalidate Caches / Restart - **did't help**

classpath 'com.android.tools.build:gradle:2.2.3'

compileSdkVersion 25
buildToolsVersion '25.0.2'

Duplicate annotation 'android.support.annotation.IntDef'; for signature: 'com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags' in the file /home/AndroidStudioProjects/constraint-layout/ContactSearch/app/build/intermediates/exploded-aar/com.takisoft.fix/preference-v7/25.1.0.1/annotations.zip!/com/takisoft/fix/support/v7/preference/annotations.xml: Duplicate annotation 'android.support.annotation.IntDef'; for signature: 'com.takisoft.fix.support.v7.preference.PreferenceFragmentCompatDividers divPrefFlags' in the file /home/denis/AndroidStudioProjects/constraint-layout/ContactSearch/app/build/intermediates/exploded-aar/com.takisoft.fix/preference-v7/25.1.0.1/annotations.zip!/com/takisoft/fix/support/v7/preference/annotations.xml

**Comment by Gericop** [comment]
@BuhTigRunner Could you create a new project (if yours is not public) that reproduces the problem? I couldn't reproduce it, so I can't really come up with a fix...
Also, what operating system do you use?

**Comment by BuhTigRunner** [comment]
@Gericop  In new project work fine. But in my project - error. 

But i fix it adding compile 'com.android.support:design:25.1.0'.  
Without this - Error: No resource found that matches the given name (at 'layout_behavior' with value '@string/**appbar_scrolling_view_behavior**')

Without library 'com.takisoft.fix', support:design library  was not needed.

**Comment by Gericop** [comment]
@BuhTigRunner Are you sure? If you check the build.gradle file of both the sample app and the lib itself, none of them includes it as a dependency, yet the sample app works perfectly. Do you have `compile "com.android.support:appcompat-v7:25.1.0"` in your build.gradle?

**Comment by BuhTigRunner** [comment]
@Gericop  Sorry to trouble you, it seems the problem is not the library, and my scattering. When I adding your lib, casually replaced  compile 'com.android.support:design:25.1.0'

**Comment by phillat** [comment]
I'm having the same problem. I have to mention that I am new to android development and working  with very limited resources. Developing on a dell inspiron mini 1018 (can't run an emulator) and testing on huawei y220 (api 10). To keep my message shorter here is summary of my environment:

1. javac version 1.8.0_102
2. Android Studio 2.2.3
3. Compile SDK 25
4. Build tools 25.0.2
5. Gradle version 2.14.1
6. Gradle plugin 2.2.3

dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    testCompile 'junit:junit:4.12'
    compile 'com.android.support:appcompat-v7:25.1.0'
    compile 'com.android.support:support-v4:25.1.0'
//    compile 'com.android.support:design:25.1.0'
    compile 'com.takisoft.fix:preference-v7:25.1.0.1'
}
Things I tried:
1. Cleaning and rebuilding project
2. Invalidating caches and restarting (error goes away but reappears after building and running)
3. Included 'com.android.support:design:25.1.0' since it worked for @BuhTigRunner

I'm learning from the udacity-google partnership courses specifically taking the android basics networking course and working on the quake report app. I can't provide you with access to my project because currently I'm experiencing slow and flaky internet connection but here is a link to the follow-up project from the course [quake report app](https://github.com/udacity/ud843-QuakeReport/tree/lesson-four). Everything is pretty much the same with exception of following:

1. Minimum sdk is set to 10 (obviously)
2. Using AsyncTaskLoader from v4 support library
3. Using PreferenceFragmentCompat from your library in my settings activity:

```
package com.example.android.quakereport;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.preference.ListPreference;
import android.support.v7.preference.Preference;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat;

public class SettingsActivity extends AppCompatActivity {
    public static final String FRAG_TAG = "Tag";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_activity);
    }

    public static class EarthquakePreferenceFragment extends PreferenceFragmentCompat
            implements Preference.OnPreferenceChangeListener {
        @Override
        public void onCreatePreferencesFix(Bundle bundle, String s) {
            //addPreferencesFromResource(R.xml.settings_main);
            setPreferencesFromResource(R.xml.settings_main, s);
        }

        @Override
        public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
            View rootView = super.onCreateView(inflater, container, savedInstanceState);

            Preference minMagnitude = getPreferenceManager()
                    .findPreference(getString(R.string.settings_min_magnitude_key));
            bindPreferenceSummaryToValue(minMagnitude);

            Preference orderBy = getPreferenceManager()
                    .findPreference(getString(R.string.settings_order_by_key));
            bindPreferenceSummaryToValue(orderBy);

            //setDividerPreferences(DIVIDER_PREFERENCE_BETWEEN | DIVIDER_CATEGORY_AFTER_LAST);
            return rootView;
        }

        private void bindPreferenceSummaryToValue(Preference preference) {
            preference.setOnPreferenceChangeListener(this);

            SharedPreferences sharedPrefs = PreferenceManager
                    .getDefaultSharedPreferences(preference.getContext());

            String preferenceString = sharedPrefs.getString(preference.getKey(), "");
            onPreferenceChange(preference, preferenceString);
        }

        @Override
        public boolean onPreferenceChange(Preference preference, Object value) {
            String stringValue = value.toString();
            if (preference instanceof ListPreference) {
                ListPreference listPreference = (ListPreference) preference;
                int prefIndex = listPreference.findIndexOfValue(stringValue);
                if (prefIndex >= 0) {
                    CharSequence[] labels = listPreference.getEntries();
                    preference.setSummary(labels[prefIndex]);
                }
            } else {
                preference.setSummary(stringValue);
            }
            return true;
        }
    }
}
```

Working with the limited resources available to me it's not just that I want to include the support library and your library but I need it because without it I couldn't even test my app on my phone. Any help would be greatly appreciated.

**Comment by Gericop** [comment]
@phillat What operating system do you use?

**Comment by Gericop** [comment]
Okay, so what I think might go wrong here is that the divider settings use two annotations: one for the base and one for the flags. Both of these annotations are annotated with `@IntDef`. However, repeating annotations were introduced to Java in the 8th version and I think that gradle / javac messes up something with this. I'm gonna merge the two together (*which means some semantical checks will not be available*) and hope for the best.

**Comment by Gericop** [comment]
Released a new version: v25.1.0.2. Please, check if the problem still exists (make sure you clear the project; also check if gradle really clears `build/intermediates/exploded-aar/com.takisoft.fix`).

**Comment by phillat** [comment]
First, thank you for your fast response. I'm using windows 7 SP 1. I will be updating to v25.0.2 as soon as my internet connection is restored to normal it's still very slow. Also, does using build->clean clear the project or should I invalidate the cache and restart? I will keep you updated once I download the new version.

**Comment by Gericop** [comment]
@phillat Make sure you update to **25.1.0.2**. Build -> clean should be enough.

**Comment by phillat** [comment]
One more thing, when I included a switch in my settings activity (just experimenting) using SwitchPreferenceCompat the switch_thumb is not properly centered on the switch_track. I installed the app on a couple of my friends' phones ranging from api 16 - api 21 and it looks great and perfectly aligned like it's supposed to. I don't have access to another phone running api 10 on which to test the app in order to be sure if it's got to do with my phone or not. But from your readme you said you tested the library on api 10, did you notice anything like that?

**Comment by phillat** [comment]
Thank you, I will update the library ASAP and get back to you.

**Comment by Gericop** [comment]
@phillat Could you open a new issue with a screenshot attached? I'm not sure about the problem.

**Comment by phillat** [comment]
Okay, I will in a couple of minutes because it'll take a while for the image to be uploaded. Just to give you an idea of what I'm dealing with it takes almost a minute and half for my comments to be submitted.

**Comment by phillat** [comment]
Finally, I was able to update the library to v25.0.2 and so far everything seems fine. I'm still testing it though will update you of any possible errors (I hope not though). In the meantime I'm going to enjoy using your library. Thank you for your help and awesome library!

**Comment by eriknyk** [comment]
still happening even on v7:27.1.1

```
implementation 'com.android.support:support-v4:27.1.1'
    implementation 'com.android.support:support-v13:27.1.1'
    implementation 'com.android.support:cardview-v7:27.1.1'
    implementation 'com.android.support:appcompat-v7:27.1.1'
    implementation 'com.android.support:design:27.1.1'
    implementation ('com.android.support:preference-v7:27.1.1') {
```

error:
```
AGPBI: {"kind":"error","text":"error: duplicate value for resource \u0027attr/maxWidth\u0027 with config \u0027\u0027.","sources":[{"file":"/Users/erik/.gradle/caches/transforms-1/files-1.1/appcompat-v7-27.1.1.aar/de16534766b25cf86037771b3f3fc3f7/res/values/values.xml","position":{"startLine":250,"startColumn":4,"startOffset":27058,"endColumn":68,"endOffset":27122}}],"original":"","tool":"AAPT"}
AGPBI: {"kind":"error","text":"error: resource previously defined here.","sources":[{"file":"/Users/erik/.gradle/caches/transforms-1/files-1.1/appcompat-v7-27.1.1.aar/de16534766b25cf86037771b3f3fc3f7/res/values/values.xml","position":{"startLine":250,"startColumn":4,"startOffset":27058,"endColumn":68,"endOffset":27122}}],"original":"","tool":"AAPT"}
:mergeDebugResources
Error: java.util.concurrent.ExecutionException: com.android.builder.internal.aapt.v2.Aapt2Exception: AAPT2 error: check logs for details
:mergeDebugResources FAILED

```
